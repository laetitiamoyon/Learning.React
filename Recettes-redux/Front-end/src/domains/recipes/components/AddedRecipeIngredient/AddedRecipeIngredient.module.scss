@import '../../../../styles/styles.scss'; 

.li { display: flex; padding-top: 18px; @include text(undefined, $tertiaryColor);}

.trashIcon
{
    @include superBackground(url(../images/icons/trash.png), cover);
    @include widthHeight(22px, 22px);
    margin-left : $spacingXs;

    cursor: pointer;
}