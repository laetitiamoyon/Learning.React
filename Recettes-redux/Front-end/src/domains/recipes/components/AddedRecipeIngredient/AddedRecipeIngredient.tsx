import React from 'react';
import { FC } from 'react';
import { RecipeIngredients } from '../../recipes.model';
import styles from './AddedRecipeIngredient.module.scss';

interface Props extends RecipeIngredients
{
  removeAddedIngredient : (id : string) => void
}

const AddedRecipeIngredient : FC<Props> = ({id, title, quantity, unity, removeAddedIngredient}) => 
{
  const removeIngredient = () => removeAddedIngredient(id)
  
  return <li className={styles.li}>
      - {quantity} {unity} de {title}
      <div className={styles.trashIcon} onClick={removeIngredient}></div>
    </li>
}


export default AddedRecipeIngredient;