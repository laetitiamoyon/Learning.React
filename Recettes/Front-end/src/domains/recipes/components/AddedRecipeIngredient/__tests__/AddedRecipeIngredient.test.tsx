import {render} from "@testing-library/react";
import AddedRecipeIngredient from "../AddedRecipeIngredient";

const mockRemoveIngredient = jest.fn()

describe("AddedRecipeIngredient", () =>
{
    it("should match snapshot", () =>
    {
        //Given & When
        const { container } = render(<AddedRecipeIngredient
            removeAddedIngredient={""} id={''} unity={''} title={''} quantity={undefined} />)

        //Then
        expect(container).toMatchSnapshot()
    })
})