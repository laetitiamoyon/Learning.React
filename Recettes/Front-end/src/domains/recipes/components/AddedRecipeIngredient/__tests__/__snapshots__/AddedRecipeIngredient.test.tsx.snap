// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddedRecipeIngredient should match snapshot 1`] = `
<div>
  <li
    class="li"
  >
    - 
     
    
     de 
    
    <div
      class="trashIcon"
    />
  </li>
</div>
`;
